src/Servant/QuickCheck/Internal/Predicates.hs

```diff
@@ -84,15 +84,13 @@ notLongerThan maxAllowed
 -- /Since 0.0.0.0/
 onlyJsonObjects :: ResponsePredicate
 onlyJsonObjects
-  = ResponsePredicate (\resp -> case go resp of
-      Nothing -> throw $ PredicateFailure "onlyJsonObjects" Nothing resp
-      Just () -> return ())
-    where
-      go r = do
-        ctyp <- lookup "content-type" (first foldedCase <$> responseHeaders r)
-        when ("application/json" `SBS.isPrefixOf` ctyp) $ do
-          (_ :: Object) <- decode (responseBody r)
-          return ()
+  = ResponsePredicate (\resp -> do
+        case lookup "content-type" (first foldedCase <$> responseHeaders resp) of
+            Nothing    -> return ()
```

Comment:
If we don't have a content-type, should we verify that the body *isn't* a JSON object?

Reply:
Well the documentation for onlyJsonObjects states it's intentions as stopping people from returning JSON that isn't wrapped in an object. If there is no content-type, I would assume we are supposed (from the API user POV) to ignore anything in the body, if there is one at all? This mostly occurs in errors... Checking that responses with a certain content have the correct content-type header (such as, having one) would be more logically placed in another predicate IMHO ?